BigBaseV2/src/views/vehicle/view_spawn.cpp

```diff
@@ -24,66 +15,73 @@ namespace big
-			const auto veh = vehicle::spawn(model, location, g_local_player->m_player_info->m_ped->m_navigation->m_right.x + 90.f);
+		ImGui::SetNextItemWidth(300.f);
+		components::input_text_with_hint("Plate", "Plate Number", plate, sizeof(plate), ImGuiInputTextFlags_None, [] {
+			g->spawn.plate = plate;
+		});
 
-			if (g->spawn.spawn_inside)
-				PED::SET_PED_INTO_VEHICLE(PLAYER::PLAYER_PED_ID(), veh, -1);
+		static char search[64];
+		static std::string lower_search;
 
-			if (g->spawn.spawn_maxed)
-				vehicle::max_vehicle(veh);
+		ImGui::SetNextItemWidth(300.f);
+		components::input_text_with_hint("Model Name", "Search", search, sizeof(search), ImGuiInputTextFlags_None, [] {
+			lower_search = search;
+			std::transform(lower_search.begin(), lower_search.end(), lower_search.begin(), tolower);
 		});
+
 		// arbitrary subtraction this looked nice so idc, works for all resolutions as well
-		if (ImGui::ListBoxHeader("###vehicles", { 0, static_cast<float>(*g_pointers->m_resolution_y - 260)}))
-		{
-			if (!g_vehicle_preview_service->get_vehicle_list().is_null())
-			{
-				for (auto& item : g_vehicle_preview_service->get_vehicle_list())
-				{
-					if (item["Name"].is_null() || item["DisplayName"].is_null())
-						continue;
-
-					std::string name = item["Name"];
-					std::string display_name = item["DisplayName"];
-
-					std::string manufacturer;
-					std::string search = model;
-					std::transform(search.begin(), search.end(), search.begin(), ::tolower);
-
-					if (!item["ManufacturerDisplayName"].is_null())
-						manufacturer = item["ManufacturerDisplayName"];
-
-					if (search.empty() ||
-						does_search_match(name, search) ||
-						does_search_match(display_name, search) ||
-						does_search_match(manufacturer, search))
-					{
-						components::selectable(item["DisplayName"], item["Name"] == search, [&item]
-						{
-							const auto location = self::pos;
-							const Vehicle veh = vehicle::spawn(item["Name"], location, 0.f);
-
-							if (g->spawn.spawn_inside)
-							{
+		if (ImGui::ListBoxHeader("###vehicles", { 300, static_cast<float>(*g_pointers->m_resolution_y - 260)})) {
+			if (g_vehicle_preview_service->get_vehicle_map().size() > 0) {
+
+				for (auto& [hash, item] : g_vehicle_preview_service->get_vehicle_map()) {
+					std::string display_name = item.dispaly_name;
+					std::string display_manufacturer = item.display_manufacturer;
+
+					std::transform(display_name.begin(), display_name.end(), display_name.begin(), ::tolower);
+					std::transform(display_manufacturer.begin(), display_manufacturer.end(), display_manufacturer.begin(), ::tolower);
+
+					if (
+						display_name.find(lower_search) != std::string::npos ||
+						display_manufacturer.find(lower_search) != std::string::npos
+					) {
+						std::string& item_name = item.name;
+
+						components::selectable(item.dispaly_name + "##" + std::to_string(item.hash), false, [&item_name] {
```

Comment:
ImGui::PushId() can be used instead of using string concatenation.

Don't forget to ImGui::PopId() after the selectable.

Reply:
Fixed